cmd/execution/main.go

```diff
@@ -318,6 +329,71 @@ func main() {
 
 			return compactor, nil
 		}).
+		Component("execution data service", func(node *cmd.NodeConfig) (module.ReadyDoneAware, error) {
+			err := os.MkdirAll(executionDataDir, 0700)
+
+			if err != nil {
+				return nil, err
+			}
+
+			ds, err := badger.NewDatastore(executionDataDir, &badger.DefaultOptions)
+
+			if err != nil {
+				return nil, err
+			}
+
+			// TODO: if the node is not starting from the beginning of the spork, it may be useful to prepopulate
+			// the cache with the existing Execution Data blob trees. Currently, the cache is empty every time the
+			// node restarts, meaning that there will initially be no prioritization of requests.
+			executionDataCIDCache = state_synchronization.NewExecutionDataCIDCache(executionDataCIDCacheSize)
+
+			bs, err := node.Network.RegisterBlobService(
+				engine.ExecutionDataService,
+				ds,
+				network.WithBitswapOptions(
+					bitswap.WithTaskComparator(
+						func(ta, tb *bitswap.TaskInfo) bool {
+							ra, err := executionDataCIDCache.Get(ta.Cid)
+
+							if err != nil {
+								return false
+							}
+
+							rb, err := executionDataCIDCache.Get(tb.Cid)
+
+							if err != nil {
+								return true
+							}
+
+							if ra.BlobTreeRecord.BlockHeight > rb.BlobTreeRecord.BlockHeight {
+								// more recent block has higher priority
+								return true
+							} else if ra.BlobTreeRecord.BlockID == rb.BlobTreeRecord.BlockID {
+								// deeper node in the same blob tree has higher priority
+								return ra.BlobTreeLocation.Height < rb.BlobTreeLocation.Height
```

Comment:
is this in the right direction? Does deeper mean larger Height or smaller height?

Reply:
larger index in the array means smaller height

So the root CID would be at the last index in the array